pkg/subctl/cmd/join.go

```diff
@@ -441,6 +453,17 @@ func populateSubmarinerSpec(subctlData *datafile.SubctlData) submariner.Submarin
 	if globalCIDR != "" {
 		submarinerSpec.GlobalCIDR = globalCIDR
 	}
-
+	fmt.Printf("submariner spec is %s", submarinerSpec)
 	return submarinerSpec
 }
+
+func createSAPerCluster(clientset *kubernetes.Clientset, brokernamespace string) (*v1.Secret) {
+	broker.CreateNewBrokerSA(clientset, clusterID+"-submariner-k8s-broker-client")
+	serviceaccount, _ := clientset.CoreV1().ServiceAccounts(broker.SubmarinerBrokerNamespace).List(metav1.ListOptions{})
```

Comment:
This ignores any errors, which might give you some hint about why this is failing.